### PR TITLE
Add ${rootNameShort} template option to window.title for saved workspaces

### LINKED ISSUE #145238: Request: `${rootNameShort}` for Window Titles
On all platforms, the `${rootName}` Window Title variable includes **" (Workspace)"**, and when using Remotes like WSL, the variable also includes something like **" [WSL: Ubuntu]"**. It would be useful to have a shorter variant with just the Workspace name, so instead of **"wsName (Workspace) [WSL: Ubuntu]"** we could have **"wsName"**.

I frequently have several Workspaces open and use Workspace names to distinguish between open windows in the Windows Task Bar/View and macOS Mission Control/Exposé. The extra characters describing the Workspace often cause the OS to truncate the visible window title, hiding other useful information.  The `window.title` property allows customization, but putting the Workspace first often sacrifices filename visibility for Workspace metadata. 

Since we already have access to `${remoteName}`, perhaps other Workspace component names could also be added? 

Current:
- `${rootName}` - **wsName (Workspace) [WSL: Ubuntu]**
- `${remoteName}` - **WSL: Ubuntu**

Proposed: 
- `${rootNameShort}` - **wsName**
-  `${rootIdentifier}` - **Workspace**   _// (similar to `${separator}`_

With those two variables defined, `${rootName}` could then be expressed as `${rootNameShort} (${rootIdentifier}) [${remoteName}]`

### PR DESCRIPTION
**Pull request by Wolverine1621**
This PR fixes #145238

This PR adds a `${rootNameShort}` template option for the `window.title` property for saved workspaces. An optional `short` option is added to the `getWorkspaceName` interface and handled in the `labelService` implementation. Calls to `getWorkspaceName` with the `verbose` option, and `window.title` behavior for other types of workspaces, remain unchanged.

Example:

- Default window title: "wsName (Workspace) [Remote Name]"
- `${rootNameShort}` enabled: "wsName"

### CODE REVIEW

**Comment by Wolverine1621** [comment]
Hello! I see this has been marked as a draft - are there some things I need to review and change? If you have some time to give me some direction I'll take another look.

**Comment by Wolverine1621** [comment]
@bpasero Are there any updates on this PR currently? Thank you!

File: src/vs/workbench/browser/parts/titlebar/titlebarPart.ts [hunk 1]
```diff
@@ -326,6 +327,7 @@ export class TitlebarPart extends Part implements ITitleService {
 			activeFolderMedium,
 			activeFolderLong,
 			rootName,
+			rootNameShort,
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/titlebar/titlebarPart.ts:330]: A new variable needs to be added to here as well:

https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/extensions/configuration-editing/src/settingsDocumentHelper.ts#L81

and also here:

https://github.com/microsoft/vscode/blob/1128bdab1bcf92f643afee2af1d6c055c8475081/src/vs/workbench/browser/workbench.contribution.ts#L487

File: src/vs/workbench/browser/parts/titlebar/titlebarPart.ts [hunk 2]
```diff
@@ -310,6 +310,7 @@ export class TitlebarPart extends Part implements ITitleService {
 		const activeFolderMedium = editorFolderResource ? this.labelService.getUriLabel(editorFolderResource, { relative: true }) : '';
 		const activeFolderLong = editorFolderResource ? this.labelService.getUriLabel(editorFolderResource) : '';
 		const rootName = this.labelService.getWorkspaceLabel(workspace);
+		const rootNameShort = this.labelService.getWorkspaceLabel(workspace, { verbose: false, short: true });
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/titlebar/titlebarPart.ts:313]: This deserves a test for example in here:

https://github.com/microsoft/vscode/blob/19f6f3103606da3fe5d8be4b722d96fe8e18271a/src/vs/base/test/common/labels.test.ts#L105

**Review by bpasero — CHANGES_REQUESTED** [review]
At a quick glance I find both `verbose: boolean` and `short: boolean` hard to understand. Imho it would be easier if there was a single option `verbosity` as we do have for editor inputs:

https://github.com/microsoft/vscode/blob/0e04328299ad24b0d7ff83bc9c763fa1ac5312bc/src/vs/workbench/common/editor/editorInput.ts#L113

And then:
* `SHORT`: is just the name
* `MEDIUM`: is the default as today
* `LONG`: the same as with `verbose: true`

Besides, `rootName` does not only apply to workspaces, but also when you have a single folder opened, so I would think this needs to apply also when folders are opened. There we do not append the ` (Workspace)` suffix, but we would append the remote name.

**Review by hoaprofessional — APPROVED** [review]
good feature

**Comment by bpasero** [comment]
Closing as stale.

**Pull request closed without merging**